src/main/java/com/datastax/cdm/job/BaseJobSession.java

```diff
@@ -47,7 +47,8 @@ public abstract class BaseJobSession {
     protected RateLimiter rateLimiterOrigin;
     protected RateLimiter rateLimiterTarget;
     protected Integer maxRetries = 10;
-
+    protected Boolean trackRun = true;
```

Comment:
How much added overhead to the speed this will add? Should we make it configurable to turn it off if I don't want to  track the run?

Reply:
Its enabled by default, but its configurable & can be turned off using param `--conf spark.cdm.trackRun=false`.

**This is a good question, so elaborating a bit more:** The overhead is negligible. The computation & spliting of cluster tokens into token-ranges already happened earlier, this feature converts these calculation into a structure & persists it in DB in oneshot during the start of the run. As the run progresses by token-ranges in parallel, each parallel task will updates its own status. On an avg we expect each task to read, move & write around 10MB or data (or 10K rows of 1 KB size). While doing this the task will also do 2 updates, one to mark task `started` and one to mark task ended ('pass`, `fail`, `diff`). So the factor on an avg is ~0.01% (likely even less as these updates are only a few bytes i.e. much less than a KB). 

I tried running the jobs (Migrate & DiffData) with the same config with the old jar (no track-run feature) & new jar (with track-run feature) with ~4.2 million rows & it usually took around the same time ~3.5 mins. Sometimes the new-jar took ran faster by a couple secs & other times slower by a couple secs. i.e. there was no observed diff.

Hope this helps.